src/script/lua_api/l_mainmenu.cpp

```diff
@@ -1101,6 +1101,7 @@ int ModApiMainMenu::l_get_lan_servers(lua_State *L)
 	int top = lua_gettop(L);
 	unsigned int index = 1;
 
+	std::shared_lock lock(ServerList::lan_adv_client.mutex);
```

Comment:
Isn't mutex supposed to be unlocked at some point in this function?

Reply:
it will be unlocked in lock destructor

Reply:
Oh, ok, that makes sense.